phoenix-shared/src/commonMain/kotlin/fr.acinq.phoenix/utils/CsvWriter.kt

```diff
@@ -0,0 +1,149 @@
+
+            if (config.includesFiat) {
+                /**
+                 * Developer notes:
+                 *
+                 * - The fiat amount may not always be in the same currency.
+                 *   That is, the user can configure their preferred fiat currency in the app.
+                 *   For example:
+                 *   * user lives in USA, has currency set to USD
+                 *     * payments will record USD/BTC exchange rate at time of payment
+                 *     * exported payments will read "X.Y USD"
+                 *   * user goes on vacation in Mexico, changes currency to MXN
+                 *     * payments will record MXN/BTC exchange rate at time of payment
+                 *     * exported payments will read "X.Y MXN"
+                 *   * user moves to Spain, changes currency to EUR
+                 *     * payments will record EUR/BTC exchange rate at time of payment
+                 *     * exported payments will read "X.Y EUR"
+                 *
+                 * - Prior to February 2023, the exchange rates for fiat currencies other
+                 *   than USD & EUR may have been unreliable. So if you're parsing,
+                 *   for example COP (Colombian Pesos), and you have an alternative
+                 *   source for fetching historical exchange rates, then you may
+                 *   prefer that source over the CSV values.
+                 */
+
+                info.metadata.originalFiat?.let { originalFiatExchangeRate ->
+                    val msatsPerBitcoin = 100_000_000_000
+                    val btc = info.payment.amount.msat.toDouble() / msatsPerBitcoin.toDouble()
+                    val fiat = btc * originalFiatExchangeRate.price
+
+                    // How do we display the fiat currency value ?
+                    // Some currencies use 2 decimal places, e.g. "4.26 EUR".
+                    // But there are also zero-decimal currencies, such as JPY.
+                    //
+                    // Since it's common to have micropayments on the lightning network,
+                    // we're simply going to always display 4 decimal places.
+                    //
+                    // The extra precision can be truncated / rounded / ignored
+                    // by the reader, who has more insight into how they wish
+                    // to use the exported information.
+                    //
+                    // Note: String.companion.format isn't available until Kotlin v1.7.2
+
+                    val components = fiat.toString().split(".")
+                    val comp0 = if (components.size > 0) components[0] else "0"
+                    val comp1 = if (components.size > 1) components[1] else "0"
+
+                    val numStr = "${comp0}.${comp1.take(4).padEnd(4, '0')}"
+                    val currencyName = originalFiatExchangeRate.fiatCurrency.name
+                    val fiatStr = "$numStr $currencyName"
```

Comment:
Similar as above, the fiat value should be numeric, and the unit should be in the column header.

Reply:
I don't think that's possible, because the fiat amount may not always be in the same currency.

That is, when we store the original exchange rate, we're storing a value for 1 specific currency. Which happens to be the user's preferred fiat currency (that they configured within the app).

But the problem is that the user can change this, for whatever reason.
For example:
* user lives in USA, has currency set to USD
  * payments will record USD/BTC exchange rate at time of payment
  * exported payments will read "X.Y USD"
* user goes on vacation in Mexico, changes currency to MXN
  * payments will record MXN/BTC exchange rate at time of payment
  * exported payments will read "X.Y MXN"
* user moves to Spain, changes currency to EUR
  * payments will record EUR/BTC exchange rate at time of payment
  * exported payments will read "X.Y EUR"

Even if the user has all of the payments in the same currency, it's a bad idea to change the CSV format. Because those writing code to parse this format will make assumptions based on the samples they have on hand.